packages/server/graphql/queries/helpers/fetchGitLabProjects.ts

```diff
@@ -16,12 +14,11 @@ const fetchGitLabProjects = async (
     .get('teamMemberIntegrationAuths')
     .load({service: 'gitlab', teamId, userId})
   if (!auth?.accessToken) return []
-  const {accessToken, providerId} = auth
+  const {providerId} = auth
   const provider = await dataLoader.get('integrationProviders').load(providerId)
   if (!provider?.serverBaseUrl) return []
```

Comment:
Sometimes we've just checked whether `provider` exists and then written `provider.serverBaseUrl!` below. Personally, I prefer to include `serverBaseUrl` in the if statement as it doesn't increase the number of lines of code and it means we don't need to use a `!`

Reply:
You can also do `const provider = await dataLoader.get('integrationProviders').loadNonNull(providerId)` since you know it must exist if the `auth` was valid. I think the `teamMemberIntegrationAuths` loader should return the provider as well, since we always need it.

Reply:
`loadNonNull` guarantees that the `provider` is non-null, but the `serverBaseUrl` can still be nullable. As we have to check whether `serverBaseUrl` is null, I think there's no harm is using `if (!provider?.serverBaseUrl) return []`